src/sqlfluff/core/rules/base.py

```diff
@@ -768,36 +770,6 @@ def discard_unsafe_fixes(
                     )
                 )
 
-        # Compute the set of block IDs affected by the fixes. If it's more than
-        # one, discard the fixes. Rationale: Fixes that span block boundaries
-        # may corrupt the file, e.g. by moving code in or out of a template
-        # loop.
-        block_info = templated_file.raw_slice_block_info
-        fix_block_ids = set(block_info.block_ids[slice_] for slice_ in fix_slices)
-        if len(fix_block_ids) > 1:
-            linter_logger.info(
-                "      * Discarding fixes that span blocks: %s",
-                lint_result.fixes,
-            )
-            lint_result.fixes = []
-            return
-
-        # If the fixes touch a literal-only loop, discard the fixes.
-        # Rationale: Fixes to a template loop that contains only literals are:
-        # - Difficult to map correctly back to source code, so there's a risk of
-        #   accidentally "expanding" the loop body if we apply them.
-        # - Highly unusual (In practice, templated loops in SQL are usually for
-        #   expanding the same code using different column names, types, etc.,
-        #   in which case the loop body contains template variables.
-        for block_id in fix_block_ids:
-            if block_id in block_info.literal_only_loops:
-                linter_logger.info(
-                    "      * Discarding fixes to literal-only loop: %s",
-                    lint_result.fixes,
-                )
-                lint_result.fixes = []
-                return
-
```

Comment:
Both these checks are now obsolete. These were somewhat hacky safety mechanisms that were required when we were using the old heuristic slicer, which returned approximate results that often had errors. Now that `JinjaTracer` has replaced it, slice information is 100% accurate, and the linter has more robust mechanisms to avoid issues with fixes.